feat(react-chart): add animation on transition (#2334)
DevExpress/devextreme-reactive: packages/dx-react-chart/docs/guides/animation.md

# React Chart - Animation

You can add the [Animation](../reference/animation.md) plugin to the Chart component to animate all the series.

## Basic Setup

> Import the `Animation` plugin.

Reviewer: We often add a hyperlink to API reference

> Import the `Animation` plugin.

Reviewer: Can we add the `Related Plugins` section and one more demo to demonstrate Basic setup?

Also `Animation setups` -> `Animation Customization` or something else.

Author: We have not `Related Plugins`.

## Animation setups

You can modify animation easing and duration. For that use `duration` and `easing` properties in the `Animation` plugin. The following example demonstrates how to change easing using the d3 library and how to change duration:

.embedded-demo({ "path": "animation/custom-easing", "showThemeSelector": true })

## Transition animation

> In the following example shows animation when data updated:

Reviewer: What should I do to switch on animation on updates?
Also, can we change demo data automatically?  I think it's better.

Author: You can't to switch animation on updates. At that moment all animations is on or off.

> .embedded-demo({ "path": "animation/with-transition", "showThemeSelector": true })

Reviewer: I think this demo should be without bounce effect and display only basic animation.

# React Chart - Animation

> You can add the [Animation](../reference/animation.md) plugin to the Chart component to animate all the series.

Reviewer 2: ```suggestion
To animate chart series, add the [Animation](../reference/animation.md) plugin to the Chart component.
```

## Basic Setup

Import the [Animation](../reference/animation.md) plugin.

.embedded-demo({ "path": "animation/simple-animation", "showThemeSelector": true })

> ## Transition Animation

Reviewer 2: ```suggestion
## Animation on Transitions
```

> In the following example shows animation when data updated:

Reviewer 2: The following example shows how series are animated on data update:

.embedded-demo({ "path": "animation/with-transition", "showThemeSelector": true })

> ## Animation Customization

Reviewer 2: ```suggestion
## Customize Animation
```

> You can modify animation easing and duration. For that use `duration` and `easing` properties in the `Animation` plugin. The following example demonstrates how to change easing using the d3 library and how to change duration:

Reviewer 2: ```suggestion
Use the `Animation` plugin's `duration` and `easing` properties to specify animation's duration and easing type.

In the following code, the easing implementation is taken from the `d3` library:
```

# React Chart - Animation

To animate chart series, add the [Animation](../reference/animation.md) plugin to the Chart component.

## Basic Setup

Import the [Animation](../reference/animation.md) plugin.

.embedded-demo({ "path": "animation/simple-animation", "showThemeSelector": true })

## Animation on Transitions

The following example shows how series are animated on data update:

.embedded-demo({ "path": "animation/with-transition", "showThemeSelector": true })

## Customize Animation

> Use the `Animation` plugin's `duration` and `easing` properties to specify animation's duration and easing type.

Reviewer 3: "to specify animation's duration" a? the?

Reviewer 2: "Animation" is uncountable, so neither.

Reviewer 3: In this context, it is not. "Can I have some milk?" uncountable - no article. "The milk's colour is strange", not "milk's colour is strange". It is a possessive noun, not an uncountable noun. So, it is not neither.

Reviewer 2: Okay, then it shouldn't be a possessive noun.

```suggestion
Use the `Animation` plugin's `duration` and `easing` properties to specify animation duration and easing type.
```

Reviewer 3: Roman, 'animation', like 'text', depends on the context - they can be countable or uncountable. In this case, it should be: 
```suggestion
Use the `Animation` plugin's `duration` and `easing` properties to specify the animation's duration and easing type.
```

Reviewer 3: Or
```suggestion
Use the `Animation` plugin's `duration` and `easing` properties to specify an animation's duration and easing type.
```

Reviewer 3: an‧i‧ma‧tion /ˌænəˈmeɪʃən/ noun    
1 [uncountable] the process of making animated films, television programs, computer games etc
 They used computer animation in the film.
2 [countable] a film, television program, computer game etc that has pictures, clay models etc that seem to be really moving
